element-connector/src/main/java/org/eclipse/californium/elements/RawData.java

```diff
@@ -225,13 +227,15 @@ public static RawData inbound(byte[] data, InetSocketAddress address, Principal
 	 * 
 	 * @param data the data to send.
 	 * @param address the IP address and port the data is to be sent to.
+	 * @param correlationContext correlation context for sending data (may be <code>null</code>).
 	 * @param callback the handler to call when this message has been sent (may be <code>null</code>).
 	 * @param useMulticast indicates whether the data should be sent using a multicast message.
 	 * @return the raw data object containing the outbound message.
 	 * @throws NullPointerException if data or address is <code>null</code>.
 	 */
-	public static RawData outbound(byte[] data, InetSocketAddress address, MessageCallback callback, boolean useMulticast) {
+	public static RawData outbound(byte[] data, InetSocketAddress address, CorrelationContext correlationContext, MessageCallback callback, boolean useMulticast) {
```

Comment:
> The server then sends a response to the client using the CorrelationContext from the ongoing Exchange the response is part of, right?

Yes, that's done by this PR.

But let me short up the discussion:
I want to introduce an "extension point", means, I want to offer flexiblity outside of permanent patching californium. I sure, have not a "finished and very detailed idea" what exactly some will do with that. Therefore I would make too much assumptions and restrictions.